src/Services/OrderBooksProvider.cs

```diff
@@ -1,34 +1,36 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Threading.Tasks;
-using Common.Log;
-using Lykke.Job.OrderBooksCacheProvider.Core;
+﻿using Common.Log;
+using Lykke.Common.Log;
 using Lykke.Job.OrderBooksCacheProvider.Core.Domain;
 using Lykke.Job.OrderBooksCacheProvider.Core.Services;
-using Microsoft.Extensions.Caching.Distributed;
+using Lykke.Job.OrderBooksCacheProvider.Services.Settings;
 using Newtonsoft.Json;
+using StackExchange.Redis;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
 
 namespace Lykke.Job.OrderBooksCacheProvider.Services
 {
     public class OrderBooksProvider : IOrderBooksProvider
     {
-        private readonly IDistributedCache _cache;
+        private readonly ConnectionMultiplexer _redis;
         private readonly CacheSettings _settings;
         private readonly ILog _log;
 
-        public OrderBooksProvider(IDistributedCache cache, CacheSettings settings, ILog log)
+        public OrderBooksProvider(CacheSettings settings, ILogFactory logFactory, ConnectionMultiplexer redis)
         {
-            _cache = cache;
             _settings = settings;
-            _log = log;
+            _redis = redis;
+            _log = logFactory.CreateLog(this);
         }
 
         public async Task<IEnumerable<OrderBook>> GetCurrentOrderBooksAsync(string assetPair)
         {
             try
             {
-                var buyTask = _cache.GetStringAsync(_settings.GetOrderBookKey(assetPair, true));
-                var sellTask = _cache.GetStringAsync(_settings.GetOrderBookKey(assetPair, false));
+
+                var buyTask = _redis.GetDatabase().HashGetAsync(_settings.GetOrderBookKey(assetPair, true), "data");
+                var sellTask = _redis.GetDatabase().HashGetAsync(_settings.GetOrderBookKey(assetPair, false), "data");
```

Comment:
Here and the row above:
Database is already inited and registered in JobModule

Reply:
https://stackoverflow.com/a/25594148/9876627
>The object returned from GetDatabase is a cheap pass-thru object, and does not need to be stored.